src/python/pants/rules/core/list_target_types_test.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from contextlib import contextmanager
+from textwrap import dedent
+from typing import Optional, cast
+from unittest.mock import Mock
+
+from pants.engine.rules import UnionMembership
+from pants.engine.target import BoolField, RegisteredTargetTypes, StringField, Target
+from pants.rules.core.list_target_types import list_target_types
+from pants.testutil.engine.util import MockConsole, run_rule
+from pants.util.ordered_set import OrderedSet
+
+
+# TODO(#9141): replace this with a proper util to create `GoalSubsystem`s
+class MockOptions:
+    def __init__(self, **values):
+        self.values = Mock(**values)
+
+    @contextmanager
+    def line_oriented(self, console: MockConsole):
+        yield lambda msg: console.print_stdout(msg)
+
+
+# Note no docstring.
+class GhcVersion(StringField):
+    alias = "ghc_version"
+
+
+class HaskellLibrary(Target):
```

Comment:
What if we end up actually supporting haskell? Might get a tiny bit confusing. May be better to make up a language for this, to emphasize that this is test data.

Reply:
I'll use Fortran - I doubt we'll ever add that :smiley:

Reply:
SGTM, but fyi ;)
https://github.com/numpy/numpy/blob/b3f41eeafd2708a6b519cf3c7ba3576dfa1cc5b3/numpy/f2py/setup.py#L42-L54

Reply:
Hehe touche. I was also considering Brainfuck :)

Reply:
No obscenity in the codebase please! 😄

Reply:
https://en.wikipedia.org/wiki/Plankalk%C3%BCl

Reply:
Although Zuse was kind of a Nazi, so maybe not that either. 

ALGOL?